### PR TITLE
Refactor time service logic

### PR DESCRIPTION
**Pull request by yontyon**
Instead of having the time request as a client request inside the prePrepare message, we add it as a metadata in the prePrepare message

### CODE REVIEW

**Review by arc-vmware — APPROVED** [review]
LGTM!!!
The changes look great, thanks @yontyon for making these changes.

File: bftengine/src/bftengine/ReplicaImp.cpp [hunk 1]
```diff
@@ -696,15 +695,12 @@ PrePrepareMsg *ReplicaImp::createPrePrepareMessage() {
 
   if (config_.timeServiceEnabled) {
```
Comment (arc-vmware) [inline bftengine/src/bftengine/ReplicaImp.cpp:696]: Can we remove this check. Just return the preprepare in both cases.

File: bftengine/src/bftengine/ReplicaImp.cpp [hunk 2]
```diff
@@ -634,9 +634,8 @@ std::pair<PrePrepareMsg *, bool> ReplicaImp::buildPrePrepareMsgBatchByOverallSiz
 // So, the second value of the pair provides the real indication of success of failure.
 std::pair<PrePrepareMsg *, bool> ReplicaImp::buildPrePrepareMsgBatchByRequestsNum(uint32_t requiredRequestsNum) {
   ConcordAssertGT(requiredRequestsNum, 0);
-  // DD: To make sure that time service does not affect sending messages
-  uint32_t timeServiceBatchSizeAdjustment = config_.timeServiceEnabled ? 1 : 0;
-  if (requestsQueueOfPrimary.size() < requiredRequestsNum - timeServiceBatchSizeAdjustment) {
+
+  if (requestsQueueOfPrimary.size()) {
```
[the remark this answers is not in the document]
Comment (yontyon) [reply bftengine/src/bftengine/ReplicaImp.cpp:638]: correct, we need to fix this

File: bftengine/src/bftengine/ReplicaImp.cpp (hunk 1 above)
Comment (yontyon) [reply bftengine/src/bftengine/ReplicaImp.cpp:696]: correct

**Review by ukg19 — APPROVED** [review]
LGTM !!

**Comment by nityavasudevan** [comment]
> Changes look Good. @arc-vmware Do we directly want to merge these to 1.6?

Changes should be merged only in master for now.

**Comment by arc-vmware** [comment]
As these changes are needed only for master, I will raise the PR. Closing this PR for now.

**Pull request closed without merging**